website/docs/r/group.markdown

```diff
@@ -0,0 +1,53 @@
+---
+layout: "azuread"
+page_title: "Azure Active Directory: azuread_group"
+sidebar_current: "docs-azuread-resource-azuread-group"
+description: |-
+  Manages a Group within Azure Active Directory.
+
+---
+
+# azuread_group
+
+Manages a Group within Azure Active Directory.
+
+-> **NOTE:** If you're authenticating using a Service Principal then it must have permissions to `Read and write all groups` within the `Windows Azure Active Directory` API.
+
+-> **NOTE:** Additionally, due to a limitation within the API, you have to assign **one** of the following Azure Active Directory Roles to the Service Principal to be able to delete Groups:
+
+* User Account Administrator
+* Company Administrator
```

Comment:
after spending some time looking into this, it appears that `Company Administrator` should be the only role necessary - the `User Account Administrator` doesn't appear to have sufficient rights on it's own (I'm assuming it may for the `azuread_user` resource however?) - as such could we update this?

Reply:
`User Account Administrator` should definitely work, see https://docs.microsoft.com/en-us/azure/active-directory/users-groups-roles/directory-assign-admin-roles#user-account-administrator.

I use these permissions exclusively on my Service Principal and it seems to work.

I've found, that if you apply a role to the Service principal, it may take a few min to propagate.